File: app/src/main/java/com/example/marcin/meetfriends/ui/friends/FriendsModule.java
package com.example.marcin.meetfriends.ui.friends;

import com.example.marcin.meetfriends.ui.common.EventIdParams;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;

/**
 * Created by marci on 2017-12-03.
 */

@Module
public abstract class FriendsModule {

  @Binds
  abstract FriendsContract.View bindView(FriendsActivity view);

  @Binds
  abstract FriendsContract.Presenter bindPresenter(FriendsPresenter presenter);

  @Provides
  public static EventIdParams providesEventIdParams(FriendsActivity activity) {
    return new EventIdParams(activity.getIntent().getExtras());
  }

  @Provides
  public static ParticipantsListParams providesParticipantsListParams(FriendsActivity activity) {
    return new ParticipantsListParams(activity.getIntent().getExtras());
  }
}

File: app/src/main/java/com/example/marcin/meetfriends/ui/friends/FriendsModule.kt
package com.example.marcin.meetfriends.ui.friends

import dagger.Binds
import dagger.Module

/**
 * Created by MARCIN on 2017-11-13.
 */
@Module
abstract class FriendsModule {

  @Binds
  abstract fun bindView(view: FriendsFragment): FriendsContract.View

  @Binds
  abstract fun bindPresenter(presenter: FriendsPresenter): FriendsContract.Presenter
}